client/src/components/menuBar/macroEditor.js

```diff
@@ -0,0 +1,290 @@
+import React, { useState, useEffect } from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+import InputAdornment from '@material-ui/core/InputAdornment';
+import Icon from '@material-ui/core/Icon'
+import Slider from '@material-ui/core/Slider';
+import Grid from '@material-ui/core/Grid';
+import { OutlinedInput } from '@material-ui/core';
+
+
+const useStyles = makeStyles((theme) => ({
+  form: {
+    marginBottom: 12,
+  },
+  sliderRow: {
+    width: "100%"
+  },
+  input: {
+    width: 100,
+  },
+}));
+
+export default function MacroEditor({ goal, onChange }) {
+  const classes = useStyles();
+
+  const [calories, setCalories] = useState(goal?.calories)
+  const [protein, setProtein] = useState(goal?.protein)
+  const [carbohydrate, setCarbohydrate] = useState(goal?.carbohydrate)
+  const [fat, setFat] = useState(goal?.fat)
+
+
+  useEffect(() => {
+    if (typeof onChange === "function") {
+      const newValue = {
+        calories, protein, carbohydrate, fat
+      }
+
+      const event = new CustomEvent('change', newValue)
+      onChange(event, newValue)
+    }
+  }, [calories, protein, carbohydrate, fat])
+
+  // NOTE: lowerBound and upperBound are the lower and upper of the middle slider (carbs).
```

Comment:
Honestly this whole file feels brute forced, i.e. the interdependency between `sliders` and the `textInputs` is not elegant. 

But just keep in mind when trying to understand this in the future, 

1. There are only 2 slider bounds that will actually change, (i.e. the protein and fat sliders have one of their bounds fixed, so the only "true range slider" is the carb slider. 
1.1. `lowerBound` == lowerBound of carb slider, `upperBound` == upperBound of carb slider

2. Protein percentage = lowerBound
3. Fat percentage = 100 - upperBound


4. there are no reused methods, i.e. each macronutrient has a different method for both their slider and textInput, this was done to prevent cyclic calls


5. Changing the protein will change carb
6. Changing carb's lower bound changes protein
7. Changing carb's upper bound changes fat
8. Changing fat will change carb's upper bound